tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -441,36 +472,52 @@ private void printMembers(Map<String, Entry<Optional<GroupState>, Optional<Colle
                                 memberAssignment.host, memberAssignment.clientId, memberAssignment.numPartitions);
                         }
                         if (verbose) {
-                            String partitions;
-
-                            if (memberAssignment.assignment.isEmpty())
-                                partitions = MISSING_COLUMN_VALUE;
-                            else {
-                                Map<String, List<TopicPartition>> grouped = new HashMap<>();
-                                memberAssignment.assignment.forEach(
-                                    tp -> grouped.computeIfAbsent(tp.topic(), key -> new ArrayList<>()).add(tp));
-                                partitions = grouped.values().stream().map(topicPartitions ->
-                                    topicPartitions.stream().map(TopicPartition::partition).map(Object::toString).sorted().collect(Collectors.joining(",", "(", ")"))
-                                ).sorted().collect(Collectors.joining(", "));
-                            }
-                            System.out.printf("%s", partitions);
+                            String currentEpoch = memberAssignment.currentEpoch.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                            String currentAssignment = memberAssignment.assignment.isEmpty() ?
+                                MISSING_COLUMN_VALUE : getAssignmentString(memberAssignment.assignment);
+                            String targetEpoch = memberAssignment.targetEpoch.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                            String targetAssignment = memberAssignment.targetAssignment.isEmpty() ?
+                                MISSING_COLUMN_VALUE : getAssignmentString(memberAssignment.targetAssignment);
+                            String upgraded = memberAssignment.upgraded.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                            System.out.printf(verboseFormat, currentEpoch, currentAssignment, targetEpoch, targetAssignment, upgraded);
                         }
                         System.out.println();
                     }
                 }
             });
         }
 
-        private void printStates(Map<String, GroupInformation> states) {
+        private String getAssignmentString(List<TopicPartition> assignment) {
+            Map<String, List<TopicPartition>> grouped = new HashMap<>();
+            assignment.forEach(
+                tp -> grouped.computeIfAbsent(tp.topic(), key -> new ArrayList<>()).add(tp));
+            return grouped.entrySet().stream().map(entry -> {
+                String topicName = entry.getKey();
+                List<TopicPartition> topicPartitions = entry.getValue();
+                return topicPartitions.stream().map(TopicPartition::partition).map(Object::toString).sorted().collect(Collectors.joining(",", topicName + ":", ""));
```

Comment:
nit: This line is pretty long, don't you think? It would be great if we could break it.

```
                topicPartitions
                    .stream()
                    .map(TopicPartition::partition)
                    .map(Object::toString)
                    .sorted()
                    .collect(Collectors.joining(",", topicName + ":", ""))
```